packages/grafana-ui/src/types/datasource.ts

```diff
@@ -172,6 +173,11 @@ export abstract class DataSourceApi<
    */
   getQueryDisplayText?(query: TQuery): string;
 
+  /**
+   * Retrieve context for a given log row
+   */
+  getLogRowContext?(row: LogRowModel, limit?: number): Promise<LogRowContextQueryResponse>;
```

Comment:
This feels to Loki specific (especially response format). 

One idea would be to make it into a getLogRowContextQuery(row, limit) that returned a query, that then could be sent via the normal query method that returns SeriesData.

Reply:
I agree regarding response format. 

Crazy thought: Is this something each query row could handle/initiate if a log row is connected to a refId/query row?

Have been thinking quite a bit of where responsibilities should be in Explore, query field/editor and datasource to make Explore datasource agnostic. It's really tricky and have no good answers. But sometime it feels like maybe some concept is missing between query fields and datasource. 

Feels like this is related to being able to see logs given a metrics series or the other way around, see metrics given a log row. Not something to take into account now, but something to think about moving forward.

Reply:
> One idea would be to make it into a getLogRowContextQuery(row, limit) that returned a query, that then could be sent via the normal query method that returns SeriesData.

This one is tricky actually, taking the specifics of loki query into account. I was thinking about utilising `query` method to retrieve the context, but context query is a very limited one(time range, target and limit only), while `DataQueryRequest` interface is way more reacher. And adjusting the context query to that interface just felt wrong. 

But I agree about the response format that it should be SeriesData. I'm gonna adjust that!